console/command/extension/events.php

```diff
@@ -0,0 +1,510 @@
+	{
+		if (!defined('TITANIA_CONTRIBS_TABLE'))
+		{
+			include($root_path . 'ext/phpbb/titania/common.' . $php_ext);
+		}
+
+		// Set up the injected properties
+		$this->language = $language;
+		$this->db = $db;
+		$this->root_path = $root_path;
+
+		// Save our table names
+		$this->contribs_table = TITANIA_CONTRIBS_TABLE;
+		$this->revisions_table = TITANIA_REVISIONS_TABLE;
+		$this->attachments_table = TITANIA_ATTACHMENTS_TABLE;
+
+		// Set up empty structure
+		$this->event_listing = [
+			'template' 	=> [],
+			'php' 		=> [],
+		];
+
+		// Custom events
+		$this->custom_php_events = [];
+
+		// Temporary folder
+		$this->tmp_folder = $this->root_path . 'ext/phpbb/titania/files/contrib_temp/tmp';
+
+		$language_files = ['console'];
+		$this->language->add_lang($language_files, 'phpbb/titania');
+
+		parent::__construct($user);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	protected function configure()
+	{
+		$this->setName(self::COMMAND_NAME)
+			->setDescription($this->language->lang('CLI_DESCRIPTION_EXTENSION_EVENTS'));
+	}
+
+	/**
+	 * Execute the script
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int|null|void
+	 * @throws \Exception
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output)
+	{
+		// Store the IO object for use later
+		$this->input = $input;
+		$this->output = $output;
+
+		// Show quick explanation
+		$this->output->writeln($this->language->lang('CLI_DESCRIPTION_EXTENSION_EVENTS_EXPLAIN'));
+
+		// Run the script - could take a while if there's a lot of extensions to unzip and scan
+		$start = time();
+		$count = $this->analyse();
+		$total = time() - $start;
+
+		// Show the execution time
+		$total_time = $this->execution_time($total);
+		$average_time = $this->execution_time($total, $count);
+
+		$this->output->writeln($this->language->lang('CLI_EXTENSION_EVENTS_EXECUTION_TIME',
+			$total_time['h'], $total_time['m'], $total_time['s'], $average_time['h'], $average_time['m'], $average_time['s']));
+	}
+
+	/**
+	 * Scan the files for events
+	 * @return int
+	 * @throws \Exception
+	 */
+	private function analyse()
+	{
+		// Get the latest approved revision of each extension
+		$revisions = $this->get_latest_approved_revisions();
+
+		$progress = $this->create_progress_bar(
+			count($revisions),
+			new SymfonyStyle($this->input, $this->output),
+			$this->output,
+			true
+		);
+
+		// Start the progress bar
+		$progress->setMessage($this->language->lang('CLI_EXTENSION_EVENTS_SCAN_START'));
+		$progress->start();
+
+		// Remove temp folder - clean up before we begin
+		$this->remove_temporary_folder();
+
+		foreach ($revisions as $revision)
+		{
+			// Increase the progress
+			$progress->setMessage($this->language->lang('CLI_EXTENSION_EVENTS_SCAN_CONTRIB', $revision['contrib_name'], $revision['revision_id'], $revision['real_filename']));
+			$progress->advance();
+
+			// Read the attachment
+			if ($revision['extension'] == 'zip')
```

Comment:
Done. For my benefit, what is the rule of thumb with these situations? 
Only use `==` if the value could be of multiple types, like checking a number matches some value which was supplied in POST text?

Reply:
basically yes. in this case we're comparing against a string of text, so we can use the strict method since it won't be questionable, like if we were comparing a sting number against an integer. I only use the less strict comparisons when I'm not sure if a value might be `"1"` or `1`, for example.